opbeat.go

```diff
@@ -0,0 +1,443 @@
+func (opbeat *Opbeat) CaptureMessage(message string, l Level) error {
+	p, err := newPacket(message, nil)
+	if err != nil {
+		return err
+	}
+
+	p.Level = l
+
+	opbeat.queue(p)
+
+	return nil
+}
+
+// Waits for all packets to send.
+func (opbeat *Opbeat) Wait() {
+	opbeat.wait.Wait()
+}
+
+// Starts a goroutine which listens on the main channel and sends any packets it
+// receives to Opbeat. The goroutine will exit if the client's `.Close` function
+// is called.
+func (opbeat *Opbeat) Start() {
+	opbeat.packets = make(chan *packet)
+	go func() {
+		var p *packet
+		var open bool
+		for {
+			select {
+			case p, open = <-opbeat.packets:
+				if !open {
+					return
+				}
+				err := opbeat.send(p)
+				if err != nil {
+					opbeat.Logger.Println(err)
+				}
+				opbeat.wait.Done()
+			}
+		}
+	}()
+}
+
+// Waits for all requests to finish and closes the main channel. Closing the
+// channel will force the goroutines communicating packets to return.
+func (opbeat *Opbeat) Close() {
+	opbeat.Wait()
+	close(opbeat.packets)
+}
+
+func (opbeat *Opbeat) queue(p *packet) {
+	opbeat.wait.Add(1)
+	opbeat.packets <- p
+}
+
+func (opbeat *Opbeat) send(p *packet) error {
+	body, err := json.Marshal(p)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("https://%s/api/v1/organizations/%s/apps/%s/errors/",
+		opbeat.Host, opbeat.organizationId, opbeat.appId)
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", "Bearer "+opbeat.secretToken)
+
+	res, err := opbeat.Client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	body, err = ioutil.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	res.Body.Close()
+
+	switch res.StatusCode {
+	case 202:
+		if res.Header["Location"] != nil {
+			opbeat.Logger.Printf("Event details at %s", res.Header["Location"][0])
+		}
+		return nil
+	default:
+		if err != nil {
+			return err
+		}
+
+		err = fmt.Errorf("Opbeat response %v: %s", res.Status, string(body[:]))
+		return err
+	}
+
+	return nil
+}
+
+// The default client, created fresh from the environment on import. Can also be
+// used without the environment by manually configuring via the `.Credentials`
+// function and by setting fields directly on the `DefaultOpbeat` variable.
+// You may find function specific documentation on the corresponding functions
+// on the `*Opbeat` struct.
+var DefaultOpbeat = NewFromEnvironment()
+
+func Credentials(organizationId, appId, secretToken string) {
+	DefaultOpbeat.Credentials(organizationId, appId, secretToken)
+}
+
+func Handler(h http.Handler) http.Handler {
+	return DefaultOpbeat.Handler(h)
+}
+
+func CaptureError(err error) error {
+	return DefaultOpbeat.CaptureError(err)
+}
+
+func CaptureMessage(message string, l Level) error {
+	return DefaultOpbeat.CaptureMessage(message, l)
+}
+
+func Wait() {
+	DefaultOpbeat.Wait()
+}
+
+func Start() {
```

Comment:
I don't think there is any real use-case but technically you can Close it and then Start it again.